.github/workflows/code-quality-gates.yml

```
@@ -0,0 +1,395 @@
+      shared: ${{ steps.filter.outputs.shared }}
+      containers: ${{ steps.filter.outputs.containers }}
+      tests: ${{ steps.filter.outputs.tests }}
+      config: ${{ steps.filter.outputs.config }}
+      should-run-quality: ${{ steps.should-run.outputs.result }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for file changes
+        uses: dorny/paths-filter@v2
```

actionlint (1.7.7)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)